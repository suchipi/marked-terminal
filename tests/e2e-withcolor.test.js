process.env.FORCE_COLOR = 3;

import { readFileSync } from 'fs';
import { resolve, dirname } from 'path';
import { describe, it, expect } from 'vitest';
import Renderer from '../index.js';
import marked from './_marked.js';
import { fileURLToPath } from 'url';

function getFixtureFile(fileName) {
  return readFileSync(
    resolve(dirname(fileURLToPath(import.meta.url)), 'fixtures/', fileName),
    { encoding: 'utf8' }
  );
}

function markup(str) {
  var r = new Renderer();
  return marked(str, { renderer: r });
}

describe('e2e', function () {
  it('should render a document full of different supported syntax', function () {
    const actual = markup(getFixtureFile('e2e.md'));
    expect(actual).toMatchSnapshot();
  });
});
