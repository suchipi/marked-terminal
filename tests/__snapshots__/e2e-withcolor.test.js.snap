// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`e2e > should render a document full of different supported syntax 1`] = `
"[0mIt's very easy to make some words [1mbold[22m and other words [3mitalic[23m with Markdown. You can even [34mlink to Google! ([34m[4mhttp://google.com[24m[39m[34m)[39m[0m

[0mSometimes you want numbered lists:[0m

    1. [0mOne[0m
    2. [0mTwo[0m
    3. [0mThree[0m

[0mSometimes you want bullet points:[0m

    * [0mStart a line with a star[0m
    * [0mProfit![0m

[0mAlternatively,[0m

    * [0mDashes work just as well[0m
    * [0mAnd if you have sub points,[0m
      [0mput two spaces before the dash or star:
        * [0m[0mLike this[0m[0m[0m
      [0m
        * [0m[0mAnd this[0m[0m[0m
    * [0mLorem
        * [0m[0mIpsum[0m[0m[0m
      [0m
        * [0m[0mDolor[0m[0m[0m

[0mBut also,[0m

    1. [0mAnd not to forget[0m
       [0mnumbered lists
        1. [0m[0mCan also have[0m[0m[0m
       [0m
        2. [0m[0mSeveral sub points[0m[0m[0m

[35m[4m[1m# Structured documents[22m[24m[39m

[0mSometimes it's useful to have different levels of headings to structure your documents. Start lines with a [33m#[39m to create headings. Multiple [33m##[39m in a row denote smaller heading sizes.[0m

[32m[1m### This is a third-tier heading[22m[39m

[0mYou can use one [33m#[39m all the way up to [33m######[39m six for different heading sizes.[0m

[0mIf you'd like to quote someone, use the > character before the line:[0m

[90m[3m    [0mCoffee. The finest organic suspension ever devised... I beat the Borg with it.[0m[23m[39m
[90m[3m    [0m- Captain Janeway[0m[23m[39m

[0mThere are many different ways to style code with GitHub's markdown. If you have inline code blocks, wrap them in backticks: [33mvar example = true[39m.  If you've got a longer block of code, you can indent with four spaces:[0m

    [33mif (isAwesome){[39m
    [33m  return true[39m
    [33m}[39m

[0mGitHub also supports something called code fencing, which allows for multiple lines without indentation:[0m

    [33mif (isAwesome){[39m
    [33m  return true[39m
    [33m}[39m

[0mAnd if you'd like to use syntax highlighting, include the language:[0m

    [36mif[39m (isAwesome){
      [36mreturn[39m [36mtrue[39m
    }

"
`;
